<launch>
    <node name="controller" pkg="pid" type="controller" ns="left_wheel"   >
      <param name="node_name" value="left_wheel_pid" />
      <remap from="pid_enable" to="left_wheel_pid_activate" />
      <param name="Kp" value="40.0" />
      <param name="Ki" value="1" />
      <param name="Kd" value="5" />
      <param name="upper_limit" value="60" />
      <param name="lower_limit" value="-60" />
      <param name="windup_limit" value="60" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="10.0" />
      <param name="min_loop_frequency" value="10.0" />
      <param name="min_loop_frequency" value="10.0" />
      <remap from="setpoint" to="/setpoint_left" />
     </node>

     <node name="controller" pkg="pid" type="controller" ns="right_wheel"   >
      <param name="node_name" value="right_wheel_pid" />
      <remap from="pid_enable" to="right_wheel_pid_activate" />
      <param name="Kp" value="40.0" />
      <param name="Ki" value="1" />
      <param name="Kd" value="5" />
      <param name="upper_limit" value="60" />
      <param name="lower_limit" value="-60" />
      <param name="windup_limit" value="60" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="10.0" />
      <param name="min_loop_frequency" value="10.0" />
      <remap from="setpoint" to="/setpoint_right" />
     </node>

     <node name="controller" pkg="pid" type="controller" ns="front_wheel"   >
      <param name="node_name" value="front_wheel_pid" />
      <remap from="pid_enable" to="front_wheel_pid_activate" />
      <param name="Kp" value="40.0" />
      <param name="Ki" value="1" />
      <param name="Kd" value="5" />
      <param name="upper_limit" value="60" />
      <param name="lower_limit" value="-60" />
      <param name="windup_limit" value="60" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="10.0" />
      <param name="min_loop_frequency" value="10.0" />
      <remap from="setpoint" to="/setpoint_front" />
     </node>

     <node name="controller" pkg="pid" type="controller" ns="back_wheel"   >
      <param name="node_name" value="back_wheel_pid" />
      <remap from="pid_enable" to="back_wheel_pid_activate" />
      <param name="Kp" value="40.0" />
      <param name="Ki" value="1" />
      <param name="Kd" value="5" />
      <param name="upper_limit" value="60" />
      <param name="lower_limit" value="-60" />
      <param name="windup_limit" value="60" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="10.0" />
      <param name="min_loop_frequency" value="10.0" />
      <remap from="setpoint" to="/setpoint_back" />
     </node> 

     <node pkg="hermesIII" type="send_velocity.py" name="send_velocity"/>

     <node pkg="hermesIII" type="PID_general.py" name="PID_general"/>

     <node pkg="hermesIII" type="fix_integralError.py" name="fix_integralError"/>

   <!-- rqt_plot is a resource hog, so if you're seeing high CPU usage, don't launch rqt_plot  -->
    <!--<node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"
    args="/left_wheel/control_effort/data /left_wheel/state/data /setpoint_left/data
    /right_wheel/control_effort/data /right_wheel/state/data /setpoint_right/data
    /front_wheel/control_effort/data /front_wheel/state/data /setpoint_front/data
    /back_wheel/control_effort/data /back_wheel/state/data /setpoint_back/data" /> -->


   <!-- <node name="rqt_robot_monitor" pkg="rqt_robot_monitor" type="rqt_robot_monitor" /> -->
   <!-- <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" /> -->
</launch>
